### ISSUE TITLE
Setting PDO::ATTR_STRINGIFY_FETCHES throws exception

### ISSUE THREAD

**Issue by SakiTakamachi** [microsoft/msphpsql#1474]
It is not reasonable to communicate forever in the PR that have already been merged, so I have created a new issue.
Reference: https://github.com/microsoft/msphpsql/pull/1468

I've been working on this issue for almost a month now and the PR that fixes the issue was already merged into the dev branch 3 weeks ago.
What's stopping you from releasing it?

I'm sure all contributors willing to help this repository will do so once it's clear what the obstacles are that are preventing a release.

**PHP version**  
PHP 8.1.22+ or 8.2.9+ or 8.3.0+

**PHP SQLSRV or PDO_SQLSRV version**  
All version

**Problem description**  
Due to modifications made to PDO Core's `set_attributes` function, attempting to set `PDO::ATTR_STRINGIFY_FETCHES` will throw an exception.

**Expected behavior and actual behavior**  
I was expecting no exception to be thrown.

Actual result:
```
Fatal error: Uncaught PDOException: SQLSTATE[IMSSP]: An invalid attribute was designated on the PDO object. in /var/www/html/test.php:5
Stack trace:
#0 /var/www/html/test.php(5): PDO->setAttribute(17, false)
#1 {main}
  thrown in /var/www/html/test.php on line 5
```

**Repro code or steps to reproduce**
```
<?php

// my env
$pdo = new PDO('sqlsrv:server=mssql-server;database=test;', 'test_user', 'p@ssw0rd');
$pdo->setAttribute(PDO::ATTR_STRINGIFY_FETCHES, false);
```

**Comment by SakiTakamachi** [comment]
@David-Engel 
I've opened an issue, so please respond here.
We've been left alone for about a week, so I'm hoping for some response soon.

**Comment by youkidearitai** [comment]
@David-Engel Upstream(php-src) rejected to your opinion and we should fight to Microsoft pressure. You should release new version that include this PR.
related: https://github.com/php/php-src/issues/12038

**Comment by David-Engel** [comment]
Had a discussion with the team the other day. We'll release a hotfix to accommodate the behavior change made upstream. The original fix needs some adjustment, though. We only release PHP minor-version-specific binaries, so the compile time #if logic needs adjustment. @absci is working on the release.

**Comment by SakiTakamachi** [comment]
@David-Engel 
I understand the situation. Thank you very much for your efforts in implementing the hotfix.

I am very relieved to receive this news.
We are looking forward to the release.

**Comment by absci** [comment]
Hi, I'll be working on the hotfix release.

**Comment by SakiTakamachi** [comment]
@absci 
Thank you. This is truly great news.

**Comment by youkidearitai** [comment]
@absci 
@David-Engel 
Thanks very much for moving forward and I'm sorry to using strong words.

**Comment by qqzm** [comment]
Hi,
Is this hotfix available yet? If not, what is the timescale?
How do we install it?

Thanks,
Stephen

**Comment by absci** [comment]
The new release should be ready by the end of this week.

**Comment by agould** [comment]
Will there be a 5.10.2 release to resolve this issue? My company is unfortunately stuck on SQL Server 2012 so I cannot use 5.11.x.

**Comment by David-Engel** [comment]
> Will there also be a 5.10.x release that resolves this issue? My company is unfortunately stuck on SQL Server 2012 so I cannot use 5.11.x.

@agould

There are no plans to backport the change to 5.10.

Have you tried 5.11 against SQL Server 2012? Since most of the functionality is in the underlying ODBC driver and we haven't made significant changes from 5.10 to 5.11, I would be surprised if 5.11 didn't work fine against it. I realize 5.11 doesn't technically "support" SQL 2012, but that's mainly because we stop testing against SQL versions that are out of support.

**Comment by agould** [comment]
@David-Engel Thanks for the response. I have not tried 5.11. I will give it a try in one of our test environments.

One thing I did try this morning was removing the ATTR_STRINGIFY_FETCHES attribute:

`            $dbh->setAttribute(PDO::ATTR_ERRMODE, PDO::ERRMODE_EXCEPTION);
            //$dbh->setAttribute(PDO::ATTR_STRINGIFY_FETCHES, false);
            $dbh->setAttribute(PDO::SQLSRV_ATTR_FETCHES_NUMERIC_TYPE, true);
            $dbh->setAttribute(PDO::SQLSRV_ATTR_QUERY_TIMEOUT, 60);
`

We'd been using the first two lines since we used MySQL/MariaDB, prior to moving to SQL Server about 5 years ago, so I never thought to revisit these attributes as they just worked.

Without the ATTR_STRINGIFY_FETCHES, I don't see any negative affects. Seems the SQLSRV_ATTR_FETCHES_NUMERIC_TYPE attribute handles what we need.
